lex.go

```diff
@@ -205,6 +204,15 @@ func (l *lexer) run() {
 	for l.state = lexRule; l.state != nil; {
 		l.state = l.state(l)
 	}
+	close(l.items)
+}
+
+func (l *lexer) close() {
+	//reads all items until channel close to be sure goroutine has ended
```

Comment:
nit: space after `//` and all comments should be full sentences with a full stop
`// Reads all items... goroutine has ended.`

Reply:
Ok fixed